From f2763aaed8f421bcf266a3122a84c41a7441a5d9 Mon Sep 17 00:00:00 2001
From: John Kacur <jkacur@redhat.com>
Date: Mon, 11 Dec 2023 16:29:02 -0500
Subject: [PATCH] tuna: Don't start the gui if a display is not available

This is intended for the getopt branch of tuna

Don't start the gui if a display is not available.

Signed-off-by: John Kacur <jkacur@redhat.com>
---
 tuna-cmd.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tuna-cmd.py b/tuna-cmd.py
index 38e7017336a5..ff2920f06d6f 100755
--- a/tuna-cmd.py
+++ b/tuna-cmd.py
@@ -821,6 +821,12 @@ def main():
         sys.exit(2)
 
     if run_gui:
+        # Don't try to start the gui if no display is available
+        display = os.getenv("DISPLAY")
+        if not display:
+            usage()
+            return
+
         try:
             from tuna import tuna_gui
         except ImportError:
-- 
2.42.0

